pkg/build/builder/cmd/dockercfg/cfg.go

```diff
@@ -34,14 +39,20 @@ func (h *Helper) InstallFlags(flags *pflag.FlagSet) {
 
 // GetDockerAuth returns a valid Docker AuthConfiguration entry, and whether it was read
 // from the local dockercfg file
-func (h *Helper) GetDockerAuth(registry string) (docker.AuthConfiguration, bool) {
+func (h *Helper) GetDockerAuth(registry, authType string) (docker.AuthConfiguration, bool) {
 	var authCfg docker.AuthConfiguration
 	dockercfgPath := getDockercfgFile("")
+	if pathForAuthType := os.Getenv(authType); len(pathForAuthType) > 0 {
+		glog.Infof("%s=%s", authType, pathForAuthType)
+		dockercfgPath = getDockercfgFile(pathForAuthType)
+	}
 	if _, err := os.Stat(dockercfgPath); err != nil {
+		glog.Infof("%s: %v", dockercfgPath, err)
 		return authCfg, false
 	}
 	cfg, err := readDockercfg(dockercfgPath)
 	if err != nil {
+		glog.Infof("Error reading %s: ", dockercfgPath, err)
```

Comment:
We should then show all the output with one level or allow user to change that. Whereas the latter I don't like personally.